### PR TITLE
fix: return `/` when calling `realpathSync` with `/`

### LINKED ISSUE #863: realpathSync('/') returns invalid result (edge case bug)
This is easy to reproduce:

```js
> memfs = require('memfs');
> memfs.vol.fromJSON({'./a':'a'}, '/');
> memfs.vol.realpathSync('/')
''
```
Notice that the output is an empty string, which isn't a valid path.  This bug is caused by getPath() in node.ts:
```ts
  getPath(): string {
    return this.steps.join(SEP);
  }
```
The problem is the edge case of join when `this.steps` is `['']`, i.e., a list with a single empty string.

The following is what I would expect as output in this case:
```js
> require('fs').realpathSync('/')
'/'
```

Perhaps a good fix would be:

```ts
  getPath(): string {
    const path = this.steps.join(SEP);
    return path ? path : "/";
  }
```

### PR DESCRIPTION
**Pull request by williamstein**
-  https://github.com/streamich/memfs/issues/863

Resolves #863

### CODE REVIEW

File: src/__tests__/volume/realpathSync.test.ts [hunk 1]
```diff
@@ -12,3 +12,8 @@ describe('.realpath(...)', () => {
     expect(path).toBe('/c/index.js');
   });
 });
+
+describe("edge case -- realpathSync('/') returns '/'", () => {
```
Comment (G-Rath) [inline src/__tests__/volume/realpathSync.test.ts:16]: Similar to [your other PR](https://github.com/streamich/memfs/pull/870#discussion_r1008749448), `describe` isn't a test case - since we've already got a `describe` for `realpathSync` (though I'm not quite sure why we've got a dedicated file for it, but oh well), let's just stick the test in that with a more descriptive title:

```
describe('.realpathSync(...)', () => {
  it('works with symlinks, #463', () => {
    // ...
  });

  it('returns the root correctly', () => {
    // ...
  });
});
```

File: src/volume.ts [hunk 2]
```diff
@@ -1493,7 +1493,8 @@ export class Volume {
     const realLink = this.getResolvedLink(steps);
     if (!realLink) throw createError(ENOENT, 'realpath', filename);
 
-    return strToEncoding(realLink.getPath(), encoding);
+    const path = realLink.getPath();
+    return strToEncoding(path ? path : '/', encoding);
```
Comment (G-Rath) [inline src/volume.ts:1497]: we can just use `||` here right?

```suggestion
    return strToEncoding(realLink.getPath() || '/', encoding);
```

**Review by G-Rath — CHANGES_REQUESTED** [review]

File: src/volume.ts (hunk 2 above)
Comment (G-Rath) [reply src/volume.ts:1497]: btw fwiw this behavior technically is not correct for Windows, but I think this is still better than the current behaviour so happy to land it

image

**Comment by williamstein** [comment]
OK, I fixed these tests.

**Comment by G-Rath** [comment]
Awesome! Have you seen [this comment](https://github.com/streamich/memfs/pull/867#discussion_r1008750149)? otherwise this looks good to go

**Comment by williamstein** [comment]
I saw the comment, but I'm also not sure what to do about it, or what all of the other MS Windows conventions are with memfs, so I don't feel qualified to address it.  I may come back to it later, since Windows support is a longterm goal of my project (https://cowasm.org).

**Comment by G-Rath** [comment]
I was actually referring to the original comment/suggestion about using `||` to avoid the local variable :)

For the windows comment, thats completely fine though in future I'd recommend putting a reaction such as a thumbs up to indicate you've seen the comment and acknowledge it without any further comment yourself

**Review by G-Rath — APPROVED** [review]
